mkosi/config.py

```diff
@@ -795,6 +794,57 @@ def cache_manifest(self) -> dict[str, Any]:
         return manifest
 
 
+def parse_ini(path: Path, sections: Sequence[str] = ()) -> Iterator[tuple[str, str, str]]:
+    section = setting = value = None
+
+    for l in textwrap.dedent(path.read_text()).splitlines():
+        comment = l.find("#")
```

Comment:
This allows inline comments, which configparser does not. This brings with it the can of worms of maybe needing to escape them. 

Inline comments would be nice, but maybe for now:
```python
if l.lstrip().startswith("#"):
    continue
```

Also, configparser does allow `;` as comment syntax. Don't know if anybody ever uses it, but not having is technically a compat break.

Reply:
configparser seems to support inline comments just fine, since we're using that in quite a few places in systemd's, mkosi config

Reply:
My bad, did only a quick sanity check yesterday and
```python
c = configparser.ConfigParser()
c.read_string("""\
[DEFAULT]
foo=bar # comment
""")
c["DEFAULT"]["foo"]
```
will return the comment as part of the value, but that's because the default for `ConfigParser` is `inline_comment_prefixes=None`.